### ISSUE TITLE
Resumption of Firefox support

### ISSUE THREAD

**Issue by thomas-ashcraft** [jshackles/Enhanced_Steam_Firefox#98]
@jshackles already said that the support of the Firefox will be resumed when Mozilla will implement WebExtensions support. Unfortunately I can not find this comment. He also said that progress can be tracked on: http://arewewebextensionsyet.com/ And this site says: 
> Are we WebExtensions yet?
> Yes, I think we are.

So... is there a hope? Because I'm still using old ES for Firefox (with my own hotfixes).
Or maybe we (community) can do something to bring back ES Firefox support?

**Comment by SakhJack** [comment]
he stated in his [blog](http://www.enhancedsteam.com/blog/enhanced-steam-coming-back-to-firefox/) that there are some bugs mozilla needs to fix b4 extension can work properly 
you can use [nightly version of firefox](https://nightly.mozilla.org/) with [beta of ES](http://firefox.enhancedsteam.com/) but community pages will not be affected because of the bug

**Comment by thomas-ashcraft** [comment]
Thanks for link on blog. I didnt checked it, so i didnt know. As I see we need to wait FF version 48 to become "stable".

> Can you share what kind of hotfixes you use? Just curious.

I just unpacked the XPI and made a few fixes. Had to disable signature verification in Firefox. Do not remember exactly what I did. The main thing - I fixed community market. Issue reported here: https://github.com/jshackles/Enhanced_Steam_Firefox/issues/87
I think I even left a note about it... Yep. Here, if you're interested: 
http://pastebin.com/DJcFnQzm

**Comment by thomas-ashcraft** [comment]
Thanks for link on blog. I didnt checked it, so i didnt know. As I see we need to wait FF version 48 to become "stable".

> Can you share what kind of hotfixes you use? Just curious.

I just unpacked the XPI and made a few fixes. Had to disable signature verification in Firefox. Do not remember exactly what I did. The main thing - I fixed community market. Issue reported here: https://github.com/jshackles/Enhanced_Steam_Firefox/issues/87
I think I even left a note about it... Yep. Here, if you're interested: 
http://pastebin.com/DJcFnQzm

**Comment by garyking** [comment]
FF48 is released tomorrow, August 2. Anyone know if Enhanced Steam will work with that just fine?

**Comment by thomas-ashcraft** [comment]
FF48 is in stable. But ES almost didnt work.

Totally didnt work whole steamcommunity features.
Totally didnt work store features (with some settings).
Cofing page have white text with white backgroud.

Everyone who using stable tree of Firefox - stay at old 7.3 version of ES! Its old, but at least it works.

**Comment by thomas-ashcraft** [comment]
Actually this is disaster.
Now in Firefox 48, Add-on signing is mandatory. The override is no longer working and there is no option available to install unsigned extensions on Stable or Beta Firefox versions. And manually fixed old ES 7.3 now did not working because of this. So we left with ES 8.7 beta, which is does not work at all. AT ALL.

Screenshot of only one thing that almost working in ES 8.7 beta for Firefox:

![es87 1](https://cloud.githubusercontent.com/assets/6583664/17364485/156d7394-5989-11e6-9800-19b343030d61.png)

**Comment by jshackles** [comment]
Please be patient.  There have been a large number of fixes built specifically for Firefox that are available right now in the dev branch of this repository.  I believe these fixes will correct all or most of the issues that Enhanced Steam 8.7 exhibits in Firefox 48.

I expect that Enhanced Steam 8.8 (which will include these changes) should be released in the next few days.

**Comment by jshackles** [comment]
Actually it turns out that Firefox still has outstanding bugs and issues with their WebExtensions API documentation not matching how things actually perform in Firefox 48.  For example, the extension is given explicit permission to store.steampowered.com and steamcommunity.com domains, but is still affected by CORS issues when making requests to those domains (contrary to their documentation).

These issues seem to have been resolved in Firefox 50, so it might be a few months yet before some of these issues are totally resolved for all users.

I've updated the code for now so that Enhanced Steam will load on every page and should look correct for the most part, but when running on versions < 50, some features won't be available.
